firewood/src/merkle/proof.rs

```diff
@@ -308,10 +308,16 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
                         // If the child already resolved, then use the existing node.
                         Some(node) => merkle.get_node(node)?,
                         None => {
-                            let child_node = decode_subproof(merkle, proofs_map, &child_hash)?;
+                            let child_node_bytes = proof_nodes_map
```

Comment:
nit: can you add a comment for the block to indicate this is to decode subproof to get the node?